Updated config.md (#216)
workspacer/workspacer: docs/content/config.md

context.WorkspaceContainer.CreateWorkspace("workspace");
context.WorkspaceContainer.CreateWorkspace("another");
context.WorkspaceContainer.CreateWorkspaces("more", "than", "one");
```
 

##  Layout Engines

> LayoutEngines define the way windows get arranged on each workspace. There are a number of different ways to layout windows and you can swap between them dynamically. By default workspacer loads 2 engines to switch between, the FullLayoutEngine and the TallLayoutEngine, howeveryou can change the set of default layout engines via:

Reviewer: ```suggestion
Layout engines define the way windows get arranged on each workspace. There are a number of different ways to layout windows and you can swap between them dynamically. By default workspacer loads 2 engines to switch between, the FullLayoutEngine and the TallLayoutEngine, howeveryou can change the set of default layout engines via:
```

```csharp
context.DefaultLayouts = () => new ILayoutEngine[] { new FullLayoutEngine() };
```

 You can even manually specify the LayoutEngine(s) on a per-workspace basis, by passing the different LayoutEngines to the `CreateWorkspace` (or equivalent) function

```csharp
context.WorkspaceContainer.CreateWorkspace("layouts!", new FullLayoutEngine(), new TallLayoutEngine());
```

### Existing LayoutEngines

> * [FullLayoutEnine](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)

Reviewer: ```suggestion
* [FullLayoutEngine](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)
```

    * Maximizes the current focused window and hides all others
* [TallLayoutEngine](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)
    * Splits the screen in two horizontal zones, 1 primary and one secondary
    * Windows get created in the secondary zone by default
    * The number of windows in the primary zone can be dynamically adjusted
 * [PaneLayoutEngines](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)

>     * Orders windows in either rows or columns depending on the setting

Reviewer: I think it might be a bit clearer if `VertLayoutEngine` and `HorzLayoutEngine` are explicitly named - especially `PaneLayoutEngine` is an abstract class.

Author: good point, have addressed this in the new PR

```

### Existing LayoutEngines
* [FullLayoutEngine](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)
    * Maximizes the current focused window and hides all others
* [TallLayoutEngine](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)
    * Splits the screen in two horizontal zones, 1 primary and one secondary
    * Windows get created in the secondary zone by default
    * The number of windows in the primary zone can be dynamically adjusted
 * [PaneLayoutEngines](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Shared/Layout/PaneLayoutEngine.cs)
    * The PaneLayoutEngine is an abstract class with two implementations: the `VertLayoutEngine` and the `HorzLayoutEngine`
    * The `VertLayoutEngine` aligns windows in columns, the `HorzLayoutEngine` in rows

### In development
* GridLayoutEngine
    * Creates a dynamic (NxN) grid in which windows are sorted
    * Requires support for on-the-fly change in row and column numbers
* DwindleLayout
    * Uses primary and secondary zones as the TallLayoutEngine

>     * Tiles windows in a `left-right-bottom-left-up-right-bottom-left-up-` order

Reviewer: It's merged now! 🎉

Reviewer: Sorry I don't think I was clear - I mean that it's still under the `### In development` section

the `AddFilter` call will ensure that any windows who's title contains the text "my fun application" will be ignored by workspacer. a `true` return value will allow the window to be managed, while a `false` value will force workspacer to ignore the window. 

the `AddRoute` call will ensure that any windows who's title contains the text "Google Chrome" will be automatically placed in the "web" workspace. a null return value will signal to workspacer that the next route should be checked, while returning an actual workspace will ensure that the workspace manager will place the window in the returned workspace.

## Plugins

workspacer offers additional functionality beyond just tiling your windows through Plugins. This is just a way for a developer to ship functionality as a DLL that taps into workspacer without requiring massive amounts of extra code in your config file. Supported plugins include

>  * Menu Bar

Reviewer: ```suggestion
 * [Menu Bar](#menu-bar)
```

    * The workspacer bar creates a *nix-like* top-bar which shows the list of workspaces and other useful information
    * Widgets allow for further the customisation of the bar, look at the menu bar section for details

>  * Action Menu

Reviewer: ```suggestion
 * [Action Menu](#the-action-menu)
```

>      * The actionmenu allows the user to create custom (nested) menus which can call any function or shortcuts you'd like

Reviewer: ```suggestion
     * The action menu allows the user to create custom (nested) menus which can call any function or shortcuts you'd like
```

```

### Menu Bar Widgets

> * ActiveLayoutWidget

Reviewer: ```suggestion
* `ActiveLayoutWidget`
```

   * Shows the name of the current active WindowLayoutEngine

> * BatteryWidget

Reviewer: ```suggestion
* `BatteryWidget`
```

    * For mobile devices/laptops only
    * Shows the current battery charge percentage

>     * text color can be adjusted based on rules using the ChargeColor and ChargeTreshhold attributes , see [BatteryWidget](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Bar/Widgets/BatteryWidget.cs) for details

Reviewer: ```suggestion
    * text color can be adjusted based on rules using the ChargeColor and ChargeTreshold attributes, see [BatteryWidget](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Bar/Widgets/BatteryWidget.cs) for details
```

>  * TimeWidget

Reviewer: ```suggestion
 * `TimeWidget`
```

>     * A customisable widget to show the time with standard [C# formatting support](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings)

Reviewer: ```suggestion
    * A customizable widget to show the time with standard [C# formatting support](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings)
```

I think we should probably use American spelling for consistency.

>  * TitleWidget

Reviewer: ```suggestion
 * `TitleWidget`
```

   * Shows the title of the current focused window
   * Users can select to show just the name of the program instead of the full window title by using `isShortTitle` flag
   * More configuration details are available [here](https://github.com/workspacer/workspacer/blob/master/src/workspacer.Bar/Widgets/TitleWidget.cs)

>  * WorkspaceWidget

Reviewer: ```suggestion
 * `WorkspaceWidget`
```
